K2Bridge/Factories/AggregateFactory.cs

```diff
@@ -53,21 +54,42 @@ public static BucketAggregate GetDateHistogramAggregate(string key, DataTable da
         /// <param name="dataTable">The row collection be parsed.</param>
         /// <param name="logger">ILogger object for logging.</param>
         /// <returns><see cref="BucketAggregate"/>.</returns>
-        public static BucketAggregate GetRangeAggregate(string key, DataTable dataTable, ILogger logger)
+        public static BucketAggregate GetRangeAggregate(string key, DataTable dataTable, DataTable metadataTable, ILogger logger)
         {
+            Ensure.IsNotNull(metadataTable, nameof(DataTable));
+
             logger.LogTrace("Get range aggregate for {}", key);
 
             var rangeAggregate = new BucketAggregate() { Keyed = true };
 
+            var outputBuckets = new HashSet<string>();
+
             foreach (DataRow row in dataTable.Rows)
             {
                 var bucket = BucketFactory.CreateRangeBucket(key, row, logger);
                 if (bucket != null)
                 {
                     rangeAggregate.Buckets.Add(bucket);
+                    outputBuckets.Add(Convert.ToString(row[key]));
                 }
             }
 
+            // Get expected bucket names from metadata row
+            var metadataJson = (string)metadataTable.Rows[0][key];
```

Comment:
This code is very similar to range, can we maybe extract it to a common method?

Reply:
Done